ott/core/bar_problems.py

```diff
@@ -94,9 +94,19 @@ def segmented_y_b(
     if self._y is None or (self._y.ndim == 3 and self._b.ndim == 2):
       return self.add_slice_for_debiased(self._y, self._b)
     else:
+      if isinstance(self.cost_fn, costs.Bures):
```

Comment:
In terms of overall dependency and logic, if you insist on using a callable, I think it's better to avoid an "isinstance" if/else structure here, which might need to end up containing all possible costs. Instead, one could ask the CostFn object for a padding function by default, and not use an arg called custom_padding then. 

This could be achieved as follows: 
- By default every CostFn object would have a padding_vector method that takes dimension as input and returns a vector of zeros of that size. That method would be inherited.
- If the user uses a different cost function, such as Bures, then we could define in Bures a different, overloaded padding_vector method that takes dimension as well (this would be essentially the function you have already defined in Bures)

Reply:
Yes, that sounds perfect to me to be honest. If you agree, I would go for this solution and also move back the mean_and_cov_to_x and x_to_mean_and_cov as methods of the class Bures.

As I have mentioned, I think it is nicer to have custom segmentation (specifically padders) for each cost function. I noticed also that at Cosine there is a ridge added at init in order to avoid getting NaNs when passing a zero vector. I believe it would be nice to have custom padder for that cost too. (Also do we want to have the ridge at cosine? aka allow for zero vectors?). 

In general, I think that this custom padding may be needed again for other cost functions and maybe it is not so easy to ask the user to provide an appropriate padding vector each time. So, yes, I am more in favor of having padders as methods of the cost classes. Also, I think that custom padding (aka essentially padding with "valid" points for a given cost) will share a lot with appropriate initialization of the barycenter (which also has to be a valid point). So, I also think it would be nice to perhaps discuss initializers and padders at the same time. Pinging here @michalk8 with whom I already shared a few thoughts on this.